lib/Sema/ConstraintSystem.cpp

```diff
@@ -5886,9 +5887,28 @@ void constraints::simplifyLocator(ASTNode &anchor,
       if (!elt)
         break;
 
+      // If the 3rd element is an PackElement, add the index of pack element within 
+      // packs to locate the correct element.
+      bool hasEltPack = false;
+      unsigned eltPackIdx = 0;
+      if (path.size() > 2) {
+        auto eltPack = path[2].getAs<LocatorPathElt::PackElement>();
+        if (eltPack) {
+          hasEltPack = true;
+          eltPackIdx = eltPack->getIndex();
+        }
+      }
+
       // Extract application argument.
       if (auto *args = anchorExpr->getArgs()) {
-        if (elt->getArgIdx() < args->size()) {
+        if (hasEltPack) {
+          if (elt->getArgIdx() + eltPackIdx < args->size()) {
+            anchor = args->getExpr(elt->getArgIdx() + eltPackIdx);
+            path = path.slice(3);
+            continue;
+          }
+        }
+        else if (elt->getArgIdx() < args->size()) {
```

Comment:
This offset of pack element is originally added to avoid casting a generic pack like `1, 2, nil` into an expression when calling `SpecifyContextualTypeForNil::diagnoseAsError()`. But since I'm using `simplifyConstriant` in `fixForFole` now, the expression being diagnosed is already narrowed down to `nil`. should I still keep this?